### PR TITLE
PRESIDECMS-1963

### PR DESCRIPTION
**Pull request by Pixl8RusselCole**
View and export the answers to specific questions
View the details of an answer

### CODE REVIEW

File: system/handlers/admin/FormBuilder.cfc [hunk 1]
```diff
@@ -611,6 +653,83 @@ component extends="preside.system.base.AdminHandler" {
...
+			  questionId                = questionId
+			, startRow              = dtHelper.getStartRow()
+			, maxRows               = dtHelper.getMaxRows()
+			, orderBy               = dtHelper.getSortOrder()
+			, searchQuery           = dtHelper.getSearchQuery()
+			, sFilterExpression     = sFilterExpression
+			, savedFilterExpIdLists = savedFilterExpIdLists
+		);
+		var records = Duplicate( results.records );
+		var viewQuestionResponseTitle   = "Make this i18n";//translateResource( "formbuilder:view.submission.modal.title" );
```
Comment (sebduggan) [inline system/handlers/admin/FormBuilder.cfc]: What is "Make this i18n"? Should this have been replaced by an i18n value?

Also, remove commented-out code.

File: system/services/formbuilder/FormBuilderService.cfc [hunk 2]
```diff
@@ -946,6 +1007,149 @@ component {
...
+				, "formbuilder_question_response.website_user"
+				, "formbuilder_question_response.is_website_user"
+				, "formbuilder_question_response.admin_user"
+				, "formbuilder_question_response.is_admin_user"
+				, "formbuilder_question_response.submission"
+				, "formbuilder_question_response.submission_type"
+				, "formbuilder_question_response.submission_reference"
+				, "question.full_question_text"
+				, "submission$form.name as form_name"
+				//, "submission_context.context_name"
```
Comment (sebduggan) [inline system/services/formbuilder/FormBuilderService.cfc]: Remove commented-out blocks of code

File: system/services/formbuilder/FormBuilderService.cfc [hunk 3]
```diff
@@ -946,6 +1007,149 @@ component {
...
+				) );
+			} catch( any e ){
+
+
+			}
+		}
+
+		if ( Len( Trim( arguments.searchQuery ) ) ) {
+			extraFilters.append({
+				  //filter       = "submitted_by.display_name like :q or formbuilder_formsubmission.form_instance like :q or formbuilder_formsubmission.submitted_data like :q"
```
Comment (sebduggan) [inline system/services/formbuilder/FormBuilderService.cfc]: Remove commented-out blocks of code

File: system/preside-objects/formbuilder/formbuilder_question_response.cfc [hunk 4]
```diff
@@ -35,4 +35,8 @@ component displayname="Form builder: global question response" extends="preside.
 	property name="submission"   relationship="many-to-one" relatedto="formbuilder_formsubmission" required=false;
 	property name="website_user" relationship="many-to-one" relatedto="website_user"               required=false;
 	property name="admin_user"   relationship="many-to-one" relatedto="security_user"              required=false;
+
+	property name="is_website_user" type="boolean" formula="case when website_user > '' then 1 else 0 end";
+	property name="is_admin_user" type="boolean" formula="case when admin_user > '' then 1 else 0 end";
```
Comment (sebduggan) [inline system/preside-objects/formbuilder/formbuilder_question_response.cfc]: Should also add `dbtype` to formula fields, in this case `dbtype="boolean"`

Also, better to say `when website_user is not null`

**Review by sebduggan — CHANGES_REQUESTED** [review]
A couple of changes required here, including the removal of any commented-out blocks of code...

File: system/handlers/admin/FormBuilder.cfc (hunk 1 above)
Comment (Pixl8RusselCole) [reply system/handlers/admin/FormBuilder.cfc]: Replaced with i18n and comment removed

File: system/services/formbuilder/FormBuilderService.cfc (hunk 2 above)
Comment (Pixl8RusselCole) [reply system/services/formbuilder/FormBuilderService.cfc]: Comments removed

File: system/services/formbuilder/FormBuilderService.cfc (hunk 3 above)
Comment (Pixl8RusselCole) [reply system/services/formbuilder/FormBuilderService.cfc]: Comments removed

File: system/preside-objects/formbuilder/formbuilder_question_response.cfc (hunk 4 above)
Comment (Pixl8RusselCole) [reply system/preside-objects/formbuilder/formbuilder_question_response.cfc]: dbtype added and switched to "when website_user is not null"

**Review by sebduggan — APPROVED** [review]

**Pull request merged**